File: backend/java/src/main/java/de/iits/elocalculatorbackend/ranking/controller/RankingController.java
package de.iits.elocalculatorbackend.ranking.controller;

import de.iits.elocalculatorbackend.ranking.model.dto.RankingResponseDto;
import de.iits.elocalculatorbackend.ranking.service.RankingService;
import lombok.AllArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import java.util.List;

@RestController
@AllArgsConstructor
public class RankingController {

    private RankingService rankingService;

    @GetMapping("/rankings")
    public ResponseEntity<List<RankingResponseDto>> getRankings(@RequestParam int top) {
        return ResponseEntity.ok(rankingService.getTopPlayers(top));
    }

}

File: backend/springboot-kotlin/src/main/kotlin/de/iits/elo/ranking/controller/RankingController.kt
package de.iits.elo.ranking.controller

import de.iits.elo.ranking.model.dto.RankingResponseDto
import de.iits.elo.ranking.service.RankingService
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
class RankingController(
    private val rankingService: RankingService,
) {

    @GetMapping("/rankings")
    fun getRankings(@RequestParam top: Int): ResponseEntity<List<RankingResponseDto>> =
        ResponseEntity.ok(rankingService.getTopPlayers(top))
}
